flake.nix

```diff
@@ -0,0 +1,65 @@
+{
+  description = "Higher-order, low-boilerplate free monads.";
+
+  inputs = {
+    nixpkgs.url = github:NixOs/nixpkgs/nixos-20.09;
+    flake-utils.url = github:numtide/flake-utils;
+  };
+
+  outputs = { nixpkgs, flake-utils, ... }:
+  flake-utils.lib.eachSystem ["x86_64-linux"] (system:
+  let
+    overrides = self: super:
+    let
+      c2n = name: src: self.callCabal2nixWithOptions name src "--no-hpack" {};
```

Comment:
because we have `.cabal` checked in and it will be generated from different environments due to still using stack.
but now that I've added the source filter, we don't need it anymore!